spring-messaging/src/main/java/org/springframework/messaging/simp/broker/DefaultSubscriptionRegistry.java

```diff
@@ -411,24 +414,25 @@ public SessionInfo removeSubscriptions(String sessionId) {
 	private static final class SessionInfo {
 
 		// subscriptionId -> Subscription
-		private final Map<String, Subscription> subscriptionMap = new ConcurrentHashMap<>();
+		private final Map<String, Set<Subscription>> subscriptionMap = new ConcurrentHashMap<>();
 
-		public Collection<Subscription> getSubscriptions() {
-			return this.subscriptionMap.values();
+		public Stream<Subscription> getSubscriptions() {
+			return this.subscriptionMap.values().stream().flatMap(Set::stream);
 		}
 
-		@Nullable
-		public Subscription getSubscription(String subscriptionId) {
-			return this.subscriptionMap.get(subscriptionId);
+		public Stream<Subscription> getSubscription(String subscriptionId) {
+			return this.subscriptionMap.getOrDefault(subscriptionId, emptySet()).stream();
```

Comment:
It doesn't seem like a function that only returns one item anymore, isn't `getSubscriptions` a more appropriate name?

Reply:
It definitely is.